### PR TITLE
Email Event

### PR DESCRIPTION
**Pull request by MGatner**
**Description**
Defines a new Event, "email", triggered on successful email send. The event receives the Email class's current set of properties so listeners can respond to email specifics.

@samsonasik Please review the move of `archive` on the MockEmail class per my other comment. If you want to review this whole PR I would appreciate it :)

**Checklist:**
- [X] Securely signed commits
- [X] Component(s) with PHPdocs
- [X] Unit testing, with >80% coverage
- [X] User guide updated
- [X] Conforms to style guide

### CODE REVIEW

File: system/Email/Email.php [hunk 1]
```diff
@@ -1552,6 +1552,12 @@ public function send($autoClear = true)
 		{
 			$this->clear();
 		}
+
+		if ($result)
```
Comment (samsonasik) [inline system/Email/Email.php:1556]: if can be combined with previous if :

```php
if ($result)
{
         if ($autoClear)
         {
                   $this->clear();
         }
        
       Events::trigger('email', get_object_vars($this));

}
```

File: system/Test/Mock/MockEmail.php [hunk 2]
```diff
@@ -11,14 +12,27 @@ class MockEmail extends Email
...
+	 *
+	 * @var boolean
+	 */
+	public $returnValue = true;
+
 	public function send($autoClear = true)
 	{
+		$this->archive = get_object_vars($this);
+
 		if ($autoClear)
```
Comment (samsonasik) [inline system/Test/Mock/MockEmail.php:26]: I think it need to be : 

```suggestion
              if ($autoClear && ! $this->returnValue)
```

So the `archive` property will not be cleared when `retrunValue` is true.

File: tests/system/Email/EmailTest.php [hunk 3]
```diff
@@ -36,4 +38,42 @@ public function testEmailSendWithClearance($autoClear)
...
+			$result = $arg;
+		});
+
+		$this->assertTrue($email->send());
+
+		$this->assertIsArray($result);
+		$this->assertEquals(['foo@foo.com'], $result['recipients']);
+	}
+
+	public function testFailreDoesNotTriggerEvent()
```
Comment (samsonasik) [inline tests/system/Email/EmailTest.php:61]: typo fix: 

```suggestion
      public function testFailureDoesNotTriggerEvent()
```

File: tests/system/Email/EmailTest.php [hunk 4]
```diff
@@ -36,4 +38,42 @@ public function testEmailSendWithClearance($autoClear)
 			$this->assertEquals('foo@foo.com', $email->archive['recipients'][0]);
 		}
 	}
+
+	public function testSuccessTriggersEvent()
```
Comment (samsonasik) [inline tests/system/Email/EmailTest.php:42]: need to have consistent grammar, trigger or triggers

File: system/Email/Email.php (hunk 1 above)
Comment (MGatner) [reply system/Email/Email.php:1556]: Updated.

File: system/Test/Mock/MockEmail.php (hunk 2 above)
Comment (MGatner) [reply system/Test/Mock/MockEmail.php:26]: `clear()` does not affect `$archive` but rather a pre-defined list of properties, so I think this is fine. Your comment did make me realize though that we should only be clearing on successful returns, so I'm giving this the same consolidation treatment as `Email` above.

File: tests/system/Email/EmailTest.php (hunk 3 above)
Comment (MGatner) [reply tests/system/Email/EmailTest.php:61]: Oops! Thanks.

File: tests/system/Email/EmailTest.php (hunk 4 above)
Comment (MGatner) [reply tests/system/Email/EmailTest.php:42]: Done.

**Pull request merged**